src/main/java/me/contaria/seedqueue/SeedQueue.java

```diff
@@ -151,10 +151,31 @@ public static void discard(SeedQueueEntry entry) {
      */
     public static boolean shouldGenerate() {
         synchronized (LOCK) {
-            return getGeneratingCount() < getMaxGeneratingCount() && SEED_QUEUE.size() < config.maxCapacity;
+            return getGeneratingCount() < getMaxGeneratingCount() && hasMoreCapacity();
         }
     }
 
+    /**
+     * @return If the queue is not filled to capacity.
+     */
+    public static boolean hasMoreCapacity() {
```

Comment:
Thoughts on inverting and renaming this to `isFull`?

Reply:
Yep, since we're `!`ing it everywhere it makes sense